### PR TITLE
[FIX]Fix Infinite recursion bug in comparison script

### LINKED ISSUE #146: [BUG] Recursion error when one side is empty
Sample platform commit (found at the bottom of each page) : **Any**

**In raising this issue, I confirm the following (please check boxes, eg [X]):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that the bug-fix I am reporting can be replicated, or that the feature I am suggesting isn't already present.
- [x] I have checked that the issue I'm posting isn't already reported.
- [x] I have checked that the issue I'm porting isn't already solved and no duplicates exist in [closed issues](https://github.com/CCExtractor/sample-platform/issues?q=is%3Aissue+is%3Aclosed) and in [opened issues](https://github.com/CCExtractor/sample-platform/issues)
- [x] I have checked the pull requests tab for existing solutions/implementations to my issue/suggestion.

**My familiarity with the project is as follows (check one, eg [X]):**

- [ ] I have never visited/used the platform.
- [ ] I have used the platform just a couple of times.
- [ ] I have used the platform extensively, but have not contributed previously.
- [x] I am the benevolent dictator of this platform.

---

The comparison script that generates the html diffs between versions cannot handle it when one side is empty and throws a recursion error. A check for this should be added.

### PR DESCRIPTION
**Pull request by prabhakarpd7284**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used the project.
- [x] I have used the project briefly.
- [ ] I have used the project extensively, but have not contributed previously.
- [ ] I am an active contributor to the project.

---

Fixes #146

### CODE REVIEW

**Comment by prabhakarpd7284** [comment]
I'll make the necessary changes.

**Comment by prabhakarpd7284** [comment]
Hey @pep8speaks 
Thanks for pointing out that.
I have removed the whitespaces now. Please check.

**Comment by satyammittal** [comment]
@prabhakarpd7284 Update your branch with master.

File: mod_test/nicediff/diff.py [hunk 1]
```diff
@@ -164,7 +164,8 @@ def get_html_diff(test_correct_lines, test_res_lines):
         
     # processing remaining lines
     
-    for line in range(use+1, till):
+    for line in range(use, till):
+
```
Comment (canihavesomecoffee) [inline mod_test/nicediff/diff.py]: Why was this extra empty line added?

Comment (prabhakarpd7284) [reply mod_test/nicediff/diff.py]: It has no use. Do you want it removed?

Comment (prabhakarpd7284) [reply mod_test/nicediff/diff.py]: I have removed it.

**Comment by saurabhshri** [comment]
Did you try it with a sample? I am pretty sure this will still cause infinite recursion..

**Comment by prabhakarpd7284** [comment]
@saurabhshri 
Yes, I have checked it with samples and I am getting proper results.

**Comment by saurabhshri** [comment]
@prabhakarpd7284 I am still getting the error. Could you please tell me how are you testing it - so that I can see if I m not doing something wrong?

**Comment by prabhakarpd7284** [comment]
I am passing two strings as parameters by calling `get_html_diff()` in `diff.py`.

**Comment by saurabhshri** [comment]
Were you able to replicate the issue that way? Could you please also try by reading and trying with subtitle files?

**Comment by prabhakarpd7284** [comment]
I was able to replicate the issue.But I haven't tried with subtitle files.
I'm gonna try it now.

Btw How were you replicating the issue and testing it ?

**Comment by saurabhshri** [comment]
By passing the list containing the lines of subtitle files in `get_html_diff`. The argument's supposed to take lists.

**Comment by prabhakarpd7284** [comment]
@saurabhshri 
Please test again and check if you are still getting any error.

**Comment by saurabhshri** [comment]
Yep - looks good to me! :)

**Comment by saurabhshri** [comment]
@prabhakarpd7284 Since we're not _adding any check for empty file_ now, please update the PR title to more relevant one.

**Comment by saurabhshri** [comment]
@satyammittal please don't edit the comments that majorly changes history. If you want to make such edits please mention those in "edit section". This at least preserves sense in conversation when looked back.

**Comment by prabhakarpd7284** [comment]
@saurabhshri 
Does this title look good ?

**Pull request merged**